scripts/o3de/ExportScripts/export_utility.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+
+import o3de.command_utils as command_utils
+import o3de.export_project as exp
+import o3de.manifest as manifest
+
+from o3de import android, android_support
+
+
+import argparse
+import logging
+import pathlib
+import sys
+
+from typing import List
+
+def process_level_names(ctx, seedfile_paths, level_names, platform):
+    for level in level_names:
+        seedfile_paths.append(ctx.project_path / f'Cache/{platform}/levels' / level.lower() / (level.lower() + ".spawnable"))
+    
+def compute_tools_build_path(ctx, is_installer_sdk, selected_platform, tool_config, tools_build_path, default_base_path, logger=None):
+     # Note, when resolving the relative paths used for the export process, there is a debate about 
+    # whether to use the current working directory, or to forcefully set the base path to an O3DE directory.
+    # In practice, there was noticeable surprise whenever the relative folder wasn't found in an O3DE directory,
+    # and sometimes commands are carelessly run from sensitive directories (such as C:\\Windows\\System32)
+    
+    # For this reason, all deliberately external paths must be specified using absolute paths
+    if is_installer_sdk:    
+        tools_build_path = ctx.engine_path / 'bin' / exp.get_platform_installer_folder_name(selected_platform) / tool_config / 'Default'
+    
+    if not tools_build_path:
+        tools_build_path = default_base_path / 'build/tools'
+    elif not tools_build_path.is_absolute():
+        tools_build_path = default_base_path / tools_build_path
+    
+    if logger:
+        logger.info(f"Tools build path set to {tools_build_path}")
+    
+    return tools_build_path
+
+def handle_tools(ctx, should_build_tools, is_installer_sdk, tool_config, selected_platform, tools_build_path, default_base_path, engine_centric,logger ):
+    tools_build_path = compute_tools_build_path(ctx, is_installer_sdk, selected_platform, tool_config, tools_build_path, default_base_path)
+
+    if should_build_tools and not is_installer_sdk:
+        exp.build_export_toolchain(ctx=ctx,
+                                   tools_build_path=tools_build_path,
+                                   engine_centric=engine_centric,
+                                   tool_config=tool_config,
+                                   logger=logger)
+    
+    return tools_build_path
+
+def handle_assets(ctx, should_build_all_assets, tools_build_path, is_installer_sdk, tool_config, engine_centric, fail_on_asset_errors,
+                  ap_selected_platform, ab_selected_platform, seedlist_paths, seedfile_paths, asset_bundling_path, max_bundle_size,
```

Comment:
```suggestion
                  asset_platforms: List[str], seedlist_paths, seedfile_paths, asset_bundling_path, max_bundle_size,
```
Suggestion: Don't allow users to build asset for one platform, and then bundle for a different platform.

AssetProcessorBatch accepts "platforms" a command separated list of platforms.
`AssetProcessorBatch.exe --platforms=pc,android`

` asset_platforms_comma_separated = ','.join(asset_platforms)`

AssetBundlerBatch accepts multiple --platform args
`AssetBundlerBatch.exe --platform=pc --platform=android`

Reply:
Also, is there a way that we can tell if a platform is even enabled through the AssetProcessorSettings.setreg? By default, only `pc` is enabled, so if you are building for android, adding 'android' as a platform will not report an error, and you will end up with an empty android assets (I tested this out). The preferred process is to update a project's local `assetprocessor_settings.setreg` and enable the desired platform assets.  For example, it could look like
```
{
    "Amazon": {
        "AssetProcessor": {
            "Settings": {
                "ScanFolder Project/ShaderLib": {
                    "watch": "@PROJECTROOT@/ShaderLib",
                    "recursive": 1,
                    "order": 1
                },
                "ScanFolder Project/Shaders": {
                    "watch": "@PROJECTROOT@/Shaders",
                    "recurisve": 1,
                    "order": 2
                },
                "ScanFolder Project/Registry": {
                    "watch": "@PROJECTROOT@/Registry",
                    "recursive": 1,
                    "order": 3
                },
				"Platforms": {
				    "pc": "enabled",
					"android": "enabled"
				}
            }
        }
    }
}
```

But I don't know of any mechanism (through python at least) to check and see if an asset platform is actually enabled. This should be done BEFORE we go through the whole process of building the tools, processing the asset, etc because the user won't be able to know what went wrong.